### PR TITLE
BFD-2716: Use native migrator signaling in deployments

### PR DESCRIPTION
**Pull request by billie-rose**
**JIRA Ticket:**
[BFD-2716](https://jira.cms.gov/browse/BFD-2716)

**User Story or Bug Summary:**
As a BFD operations engineer, I want the bfd-db-migrator application to be more self-sufficient so that deployments of the migrator can be simplified and support alternative platforms.
Remove the old Migrator monitor service completely.

---

### What Does This PR Do?

Remove the dependency on the migrator monitor service in favor of the built-in SQS progress reporter.
Change the Migrator SQS queue to FIFO to guarantee the order of messages.
Update the Jenkins deployment script so that it may process the new format of the Migrator's built-in SQS, and enable said built-in SQS progress reporting by setting the appropriate environment variables.
See [this build](https://jenkins-east.cloud.cms.gov/bfd/view/Deploy/job/bfd-deploy-apps/view/change-requests/job/PR-1909/11/console) for an example of the running migrator.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] The data dictionary has been updated with any field mapping changes, if any were made.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by mackec1 — APPROVED** [review]

**Review by brianburton — CHANGES_REQUESTED** [review]
Please remove the FIFO related java changes,

I ran into specific technical issues running localstack with colima for IT tests that led @mjburling and I to remove the use of FIFO queues from SQSDao (it was there prior to that point).  Adding them back really is unnecessary, not in scope for this PR, and will add testing issues under colima.

**Comment by mjburling** [comment]
> Please remove the FIFO related java changes,

As for the AC and scope, It looks like @billie-rose has hit all the high-notes and then some! I would tend to agree. However, I am curious where this came from. @billie-rose: In your testing, did you run across cases where the lack of _guaranteed_ order in the queue was problematic, only alleviated by re-introducing the fifo queue?

> I ran into specific technical issues running localstack with colima for IT tests that led @mjburling and I to remove the use of FIFO queues from SQSDao (it was there prior to that point). Adding them back really is unnecessary, not in scope for this PR, and will add testing issues under colima.

I don't want to get _too_ carried away on support for specific alternatives like colima, but I'm _fairly_ confident this impacts other Docker Desktop alternatives. Docker Desktop is _fine_, it just requires a [license for professional use, including government use](https://docs.docker.com/subscription/desktop-license/).

**Comment by billie-rose** [comment]
@brianburton @mjburling Oh that's a bummer...I didn't realize (or remember, it was probably brought up) that fifo was an issue and had already been implemented. I did see that there were already remnants of fifo (like the `.fifo` queue name in testing) and made some false assumptions.

>  In your testing, did you run across cases where the lack of guaranteed order in the queue was problematic, only alleviated by re-introducing the fifo queue?

Only 'problematic' in the sense that the order wasn't being honored when logging to Jenkins. I thought it would be cleanest to update the implementation to maintain that order rather than mess with the logic to fit around it...I don't know the specifics limitations re: testing but it's a bummer that we can't use something like `fifo` because of test cases. I can remove it

File: ops/ansible/roles/bfd-db-migrator/defaults/main.yml [hunk 1]
```diff
@@ -5,3 +5,4 @@ db_migrator_user: bb-migrator
 db_migrator_tmp_dir: /tmp
 migrator_monitor_enabled: true
 migrator_monitor_heartbeat_interval_seconds: 300
+sqs_queue_name: bfd-test-migrator
```
Comment (lsmitchell) [inline ops/ansible/roles/bfd-db-migrator/defaults/main.yml:8]: Dont the other envs need this too?

**Review by lsmitchell — COMMENTED** [review]
Seems okay to me, just had a question about only having the sqs queue in one env it looks like?

**Review by brianburton — APPROVED** [review]
Looks good!  The groovy code is clear and easy to follow.

File: ops/ansible/roles/bfd-db-migrator/defaults/main.yml (hunk 1 above)
Comment (billie-rose) [reply ops/ansible/roles/bfd-db-migrator/defaults/main.yml]: This was actually just so the ansible tests would pass given the way we pass these environment variables. I've moved this default into `test_basic.yml` to make that a little clearer :)

**Review by lsmitchell — APPROVED** [review]

**Pull request merged**